cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/Capacity.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.model.cruisecontrol;
+
+import io.strimzi.api.kafka.model.balancing.CruiseControlBrokerCapacity;
+import io.strimzi.api.kafka.model.KafkaSpec;
+import io.strimzi.api.kafka.model.storage.EphemeralStorage;
+import io.strimzi.api.kafka.model.storage.JbodStorage;
+import io.strimzi.api.kafka.model.storage.PersistentClaimStorage;
+import io.strimzi.api.kafka.model.storage.SingleVolumeStorage;
+import io.strimzi.api.kafka.model.storage.Storage;
+
+import java.util.List;
+
+import static io.strimzi.operator.cluster.model.StorageUtils.parseMemory;
+
+public class Capacity {
+    public static final int DEFAULT_BROKER_DISK_CAPACITY   = 100000;  // in MB
+    public static final int DEFAULT_BROKER_CPU_CAPACITY    = 100;     // as a percentage (0-100)
+    public static final int DEFAULT_BROKER_NW_IN_CAPACITY  = 10000;   // in KB/s
+    public static final int DEFAULT_BROKER_NW_OUT_CAPACITY = 10000;   // in KB/s
+
+    private Integer disk;
+    private Integer cpu;
+    private Integer nwIn;
+    private Integer nwOut;
+
+    public Capacity(KafkaSpec spec) {
+        CruiseControlBrokerCapacity bc = spec.getCruiseControl().getCapacity();
+
+        if (bc != null) {
+            this.disk = bc.getDisk() != null ? bc.getDisk() : generateDiskCapacity(spec.getKafka().getStorage());
+            this.cpu = bc.getCpu() != null ? bc.getCpu() : DEFAULT_BROKER_CPU_CAPACITY;
+            this.nwIn = bc.getNwIn() != null ? bc.getNwIn() : DEFAULT_BROKER_NW_IN_CAPACITY;
+            this.nwOut = bc.getNwOut() != null ? bc.getNwOut() : DEFAULT_BROKER_NW_OUT_CAPACITY;
+        } else {
+            this.disk = generateDiskCapacity(spec.getKafka().getStorage());
+            this.cpu = DEFAULT_BROKER_CPU_CAPACITY;
```

Comment:
So this is fine if there is no user defined spec but what if a user has specified a [resource limit](https://strimzi.io/docs/latest/#resource_limits). If they have that should override the defaults in this class. 

I think you need to add a way to fetch any set resource limits and use those here if set.

Reply:
@tomncooper the user is specifying a percentage here (0-100) not an absolute value so even if he sets resource limit, this cpu capacity will be anyway a percentage on it, I don't think it really matters taking into account the resource limit absolute values here, or?

Reply:
As Paolo said, this CC CPU configuration is a percentage on the CPU resources as defined by k8s. So if a user sets a resource limit, CC can will see that value as the entire CPU and use 0-100% percentage of that CPU resource

Reply:
Good point about this being a percentage, I missed that. However the CC [BrokerCapacityInfo](https://github.com/linkedin/cruise-control/blob/master/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/BrokerCapacityInfo.java) class also contains information on the number of CPU cores:

> The information also contains the number of CPU cores and may contain disk capacities by logDirs (i.e. for JBOD).

You don't seem to include that information in this class. Are we assuming that the Broker containers will only ever have one core. I confess I am not sure if multi-core Broker containers is a thing @ppatierno? If it is and users can set resource limits on them you may need to set the limit here?

Reply:
Brokers can and should still be configured to use more than one core through the resource limits. For the CC CPU capacity configuration however, I found the additional core specification to be unnecessary:

1) Strimzi brokers are homogeneous. If X cores are specified in the Kafka resource limits, every Strimzi broker will have X cores available to it. There will be no CPU capacity difference between any of the brokers. The core specification for the CC CPU capacity configuration is intended to cover heterogeneous brokers where the CPU resources of one broker may differ from another's. The core specification adds more weight to the CPU capacity of brokers with more cores assigned to it.  

2) The CPU utilization calculation is based how actively the CPU resources are running broker threads over a set time period. For Kafka on bare metal, it is important to use the core specification for the CC CPU capacity because brokers could be sharing underlying CPU resource with other applications on the same host. The core specification details the amount of the resource the broker should be using and helps give a more accurate reading of how much of the CPU resources the broker is actually using. Since k8s handles the CPU resource isolation for us, we don't have to worry about this. Kafka Brokers on k8s will only use the CPU resources allotted by k8s resource limits. So we only need to specify the percentage of those isolated resources we want to broker to use (if not all)

Reply:
Well I'm convinced!